### ISSUE TITLE
Question about `react-you-might-not-need-an-effect/no-manage-parent`

### ISSUE THREAD

**Issue by reteps** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#21]
The following hook was flagged:

```  
export function SampleQuestionDemo({
  promptId,
  prompt,
  onMathjaxTypeset,
}: {
  promptId: keyof typeof examplePrompts;
  prompt: ExamplePrompt;
  onMathjaxTypeset: () => Promise<void>;
}) {
  const [variant, setVariant] = useState<SampleQuestionVariant | null>(null);

  // When a new variant is loaded, typeset the MathJax content.
    useLayoutEffect(() => {
      onMathjaxTypeset();
    }, [variant?.question, prompt]);
```

>   85:3  warning  This effect only uses props. Consider lifting the logic up to the parent  react-you-might-not-need-an-effect/no-manage-parent

Is the suggestion to move the state management (and hook) up a level, or is this a false positive?

**Comment by reteps** [comment]
In this case, I want it to use the `useLayoutEffect` to avoid visual glitches. However, if that was a `useEffect` instead, I only had one call site, so I could have inlined it.

> especially if that only happens in a single place. Maybe it'd be more appropriate for no-event-handler to flag this?

If your plugin could have a configurable cutoff for that depending on how many calls to `setVariant` that would be excellent. E.g. if `setVariant` is called in three locations, or passed to child components, don't flag, but if it is only called in two locations, flag.

**Comment by reteps** [comment]
In any case, feel free to close this out! Thank you for a super timely response 🚀 !
